test/e2e/reconciler/apiexportendpointslice/apiexportendpointslice_test.go

```diff
@@ -214,8 +214,14 @@ func TestAPIExportEndpointSliceWithPartitionPrivate(t *testing.T) {
 
 	t.Logf("Creating the APIExportEndpointSlice")
 	sliceClient := kcpClusterClient.ApisV1alpha1().APIExportEndpointSlices()
-	slice, err = sliceClient.Cluster(partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
-	require.NoError(t, err)
+	// allow some time for APIExport to be synced onto the cache server
+	framework.Eventually(t, func() (bool, string) {
+		slice, err = sliceClient.Cluster(partitionClusterPath).Create(ctx, slice, metav1.CreateOptions{})
+		if err != nil && !apierrors.IsAlreadyExists(err) {
```

Comment:
I don't think you'd ever get an AlreadyExists error in this flow?

Reply:
Yes I was not sure. I can remove it